library/std/src/collections/hash/set.rs

```diff
@@ -110,6 +110,7 @@ use super::map::{map_try_reserve_error, RandomState};
 /// [`Cell`]: crate::cell::Cell
 #[cfg_attr(not(test), rustc_diagnostic_item = "hashset_type")]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[doc(alias = "set")]
```

Comment:
I'm no Java expert, but isn't there `HashSet` there too? Or aliases are not supposed to be added if they are the same as the Rust name? (if yes, please see also my comment on a `lang`-like parameter: in such case I think it would make sense even if they are the same name).

Reply:
> Or aliases are not supposed to be added if they are the same as the Rust name?

Right, I don't see much point since these will show up in search anyway.

> if yes, please see also my comment on a lang-like parameter: in such case I think it would make sense even if they are the same name

What do you see as the point of doing this? I think it's fairly clear that `HashMap` will mean the same thing in Rust and Java, why annotate it? I see the goal of `doc(alias)` as making things easier to search for, which this won't help with.

Reply:
Well, I see your point of view: `alias` is used at the moment for providing search results, and since it is case-insensitive and the search already returns Rust results, then having `NULL` or `HashMap` does not make sense.

However, if you think about it from the point of view what we are doing here (the feature itself) and what it could be used for, you can easily come up with more things to do with it, and it makes sense to keep it as a mapping of other languages identifiers/features to Rust, rather than just search terms/keywords that happen to be in other languages. In which case it seems it is useful to keep track why it is an alias (i.e. the language, or the "reason" the alias is there). It is also good for documentation purposes and to keep track of them.

For instance, you could automatically generate an index of Python, or C, or Java features that are mapped to Rust for newcomers from each language, you could show the aliases in the documentation somehow (probably opt-in), you could show a small "capsule tag" with the "C" or "Python" text when searching for a term if it matched due to an alias (unless there is a Rust one exactly), etc.

So it all depends on what we are trying to do: if we are really going to keep a mapping of many functions/types from other languages/standards to Rust, then I definitely think we should add the extra metadata so that this is documented, easier to use for tooling, useful for extra features, etc.

Reply:
Rewritten as https://github.com/rust-lang/rust/issues/82000 as requested.

Reply:
I think I'd like to request dealing with this until there's an RFC for https://github.com/rust-lang/rust/issues/82000. I don't think it needs to block adding more aliases in the meantime.

Reply:
Of course! Adding `set` looks OK.